### ISSUE TITLE
Fix errors with tsc

### ISSUE THREAD

**Issue by emanuel-minetti** [emanuel-minetti/sudokuTS#7]
- [x] Fix all error messages
- [x] Adapt `README.md`

**Comment by emanuel-minetti** [comment]
As of today there were the following error messages when compiling from the cli:

```bash
 node node_modules/typescript/bin/tsc
src/lib/game/Square.ts(103,43): error TS2345: Argument of type 'number[] | null' is not assignable to parameter of type 'ArrayLike<number>'.
  Type 'null' is not assignable to type 'ArrayLike<number>'.
src/lib/solver/AbstractRules.ts(43,42): error TS2345: Argument of type '(prev: never[], curr: Square) => number[]' is not assignable to parameter of type '(previousValue: number[], currentValue: Square, currentIndex: number, array: Square[]) => number[]'.
  Types of parameters 'prev' and 'previousValue' are incompatible.
    Type 'number[]' is not assignable to type 'never[]'.
      Type 'number' is not assignable to type 'never'.
src/lib/solver/AbstractRules.ts(55,69): error TS2345: Argument of type 'number[] | null' is not assignable to parameter of type 'ArrayLike<number>'.
  Type 'null' is not assignable to type 'ArrayLike<number>'.
src/lib/solver/AbstractRules.ts(104,40): error TS2345: Argument of type 'number[] | null' is not assignable to parameter of type 'ArrayLike<number>'.
  Type 'null' is not assignable to type 'ArrayLike<number>'.
src/lib/solver/ToughRules.ts(41,55): error TS2345: Argument of type '(prev: never[], curr: Square) => number[]' is not assignable to parameter of type '(previousValue: number[], currentValue: Square, currentIndex: number, array: Square[]) => number[]'.
  Types of parameters 'prev' and 'previousValue' are incompatible.
    Type 'number[]' is not assignable to type 'never[]'.
src/lib/solver/ToughRules.ts(49,60): error TS2345: Argument of type 'number[] | null' is not assignable to parameter of type 'ArrayLike<number>'.
  Type 'null' is not assignable to type 'ArrayLike<number>'.
src/lib/solver/ToughRules.ts(66,69): error TS2345: Argument of type 'number[] | null' is not assignable to parameter of type 'ArrayLike<number>'.
  Type 'null' is not assignable to type 'ArrayLike<number>'.
```

**Comment by emanuel-minetti** [comment]
Closed by Pull Request #9
